code/game/objects/effects/anomalies.dm

```diff
@@ -232,46 +232,50 @@
-		var/obj/item/beacon/chosen
-		var/list/possible = list()
-		for(var/obj/item/beacon/W in GLOB.teleportbeacons)
-			possible += W
-
-		if(possible.len > 0)
-			chosen = pick(possible)
-
-		if(chosen)
-				// Calculate previous position for transition
-
-			var/turf/FROM = T // the turf of origin we're travelling FROM
-			var/turf/TO = get_turf(chosen) // the turf of origin we're travelling TO
-
-			playsound(TO, 'sound/effects/phasein.ogg', 100, TRUE)
-			priority_announce("Massive bluespace translocation detected.", "Anomaly Alert")
-
-			var/list/flashers = list()
-			for(var/mob/living/carbon/C in viewers(TO, null))
-				if(C.flash_act())
-					flashers += C
-
-			var/y_distance = TO.y - FROM.y
-			var/x_distance = TO.x - FROM.x
-			for (var/atom/movable/A in urange(12, FROM )) // iterate thru list of mobs in the area
-				if(istype(A, /obj/item/beacon))
-					continue // don't teleport beacons because that's just insanely stupid
-				if(A.anchored)
-					continue
-
-				var/turf/newloc = locate(A.x + x_distance, A.y + y_distance, TO.z) // calculate the new place
-				if(!A.Move(newloc) && newloc) // if the atom, for some reason, can't move, FORCE them to move! :) We try Move() first to invoke any movement-related checks the atom needs to perform after moving
-					A.forceMove(newloc)
+	if(!T)
+		return
 
-				if(ismob(A) && !(A in flashers)) // don't flash if we're already doing an effect
-					var/mob/M = A
-					if(M.client)
-						INVOKE_ASYNC(src, .proc/blue_effect, M)
+		// Calculate new position (searches through beacons in world)
+	var/obj/item/beacon/chosen
+	var/list/possible = list()
+	for(var/obj/item/beacon/W in GLOB.teleportbeacons)
+		possible += W
+
+	if(possible.len > 0)
+		chosen = pick(possible)
+
+	if(chosen)
+			// Calculate previous position for transition
+
+		var/turf/FROM = T // the turf of origin we're travelling FROM
+		var/turf/TO = get_turf(chosen) // the turf of origin we're travelling TO
+
+		playsound(TO, 'sound/effects/phasein.ogg', 100, TRUE)
+		priority_announce("Massive bluespace translocation detected.", "Anomaly Alert")
+
+		var/list/flashers = list()
+		for(var/mob/living/carbon/C in viewers(TO, null))
+			if(C.flash_act())
+				flashers += C
+
+		var/y_distance = TO.y - FROM.y
+		var/x_distance = TO.x - FROM.x
+		for (var/atom/movable/A in urange(12, FROM )) // iterate thru list of mobs in the area
+			if(istype(A, /obj/item/beacon))
+				continue // don't teleport beacons because that's just insanely stupid
+			if(A.anchored)
+				continue
+			if(HAS_TRAIT(A, TRAIT_NO_TELEPORT)) //By definition, these cannot teleport
+				continue
+
+			var/turf/newloc = locate(A.x + x_distance, A.y + y_distance, TO.z) // calculate the new place
+			if(!A.Move(newloc) && newloc) // if the atom, for some reason, can't move, FORCE them to move! :) We try Move() first to invoke any movement-related checks the atom needs to perform after moving
+				A.forceMove(newloc)
```

Comment:
I strongly recommend using `do_teleport()` and removing the anchored and no teleport checks a few lines above since they are already present in it. This is a rather old feature, that's why it still uses `forceMove()`.

Reply:
I don't see anything in do_teleport indicating it won't affect anchored things, but I can certainly implement it here and do away with the now-unnecessary trait check.

Reply:
Oh yeah, thanks for correcting me.